File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.cs
public class Solution {
    public int FindMinArrowShots(int[][] points) {
        Array.Sort(points, (a, b) => a[0].CompareTo(b[0]));
        int end = points[points.Length-1][0], cnt = 1;
        for (int i = points.Length-2; i >=0 ; i--)
        {
            if (end > points[i][1])
            {
                cnt++;
                end = points[i][0];
            }
        }

        return cnt;
    }
}

File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.cpp
class Solution {
public:
    static bool comp(vector<int> &a, vector<int> &b) {return a[1]<b[1];}
    
    int findMinArrowShots(vector<vector<int>>& points) {

        sort(points.begin(), points.end(), comp);
        
        int arrows = 1;
        int end = points[0][1];
        
        for(int i=1; i<points.size(); i++)
        {
            if(points[i][0]>end) // if x start of ballon bigger than the end of another ballon 
            {
                arrows++;
                end = points[i][1];
            }
        }
        return arrows;
    }
};